### ISSUE TITLE
Fix generator.py file loop-search

### ISSUE THREAD

**Issue by Mikkel-AI0700** [Mikkel-AI0700/ML-Custom-AI-CPP#2]
### Issue

The _write_to_file protected function's for-loop contains a logic error where the function will not remove train/test_dataset.csv files despite the actual files existing in the *_datasets_path

Loop logic reference:
`

def _write_to_file (filepath: Path, X: pd.DataFrame, Y: pd.DataFrame):
    for directory_content in os.listdir(filepath):
        if re.findall(r"(train|test)_(x|y)_dataset.csv", directory_content):
            print(f"[*] Removed: {directory_content}")

    train_x, test_x, train_y, test_y = train_test_split(
        X,
        Y,
        train_size=0.8,
        test_size=0.2,
        shuffle=True,
        random_state=42
    )

    generated_datasets = [train_x, train_y, test_x, test_y]
    filenames = ["train_x.csv", "train_y.csv", "test_x.csv", "test_y.csv"]

    for (dataset, data_filename) in zip(generated_datasets, filenames):
        dataset = pd.DataFrame(dataset)
        dataset.to_csv(os.path.join(filepath, data_filename))
`
